x/tokenfactory/simulation/sim_msgs.go

```diff
@@ -58,3 +113,14 @@ func accountCreatedTokenFactoryDenom(k keeper.Keeper, ctx sdk.Context) simtypes.
 		return iterator.Valid()
 	}
 }
+
+func getTokenFactoryDenomAndItsAdmin(k keeper.Keeper, sim *simtypes.SimCtx, ctx sdk.Context, acc legacysimulationtype.Account) (string, sdk.AccAddress, error) {
+	store := k.GetCreatorPrefixStore(ctx, acc.Address.String())
+	denoms := osmoutils.GatherAllKeysFromStore(store)
+	denom := simtypes.RandSelect(sim, denoms...)
+
+	authData, _ := k.GetAuthorityMetadata(ctx, denom)
+	admin := authData.Admin
+	addr, err := sdk.AccAddressFromBech32(admin)
+	return denom, addr, err
```

Comment:
This is useful for catching errors early
```suggestion
	addr, err := sdk.AccAddressFromBech32(admin)
	if err != nil {
	     return "", sdk.AccAddress{}, err
	}
	return denom, addr, nil
```